src/slskd/Integrations/Scripts/ScriptService.cs

```diff
@@ -64,45 +80,40 @@ private async Task HandleEvent(Event data)
             _ = Task.Run(() =>
             {
                 Process process = default;
+                var processId = Guid.NewGuid();
 
                 try
                 {
                     var run = script.Value.Run;
+                    string executable = run.Executable ?? DefaultExecutable;
+                    string args = run.Command;
 
-                    // split the string into at most 2 parts, leaving part [0] the executable and part [1] the rest of the string, but possibly null
-                    var parts = run.Split(" ", count: 2, options: StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.RemoveEmptyEntries);
-
-                    var executable = parts[0];
-                    string args = default;
-
-                    if (parts.Length > 1)
+                    if (string.IsNullOrEmpty(executable))
                     {
-                        args = parts[1].Replace("$DATA", data.ToJson());
+                        Log.Warning("Script '{Script}' will not be run: unable to determine script executable. Update the script configuration, or set your operating system's SHELL envar.");
+                        return;
                     }
 
-                    Log.Debug("Running script '{Script}': {Run}", script.Key, run);
+                    Log.Debug("Running script '{Script}': \"{Executable}\" {Args} (id: {ProcessId})", script.Key, executable, args, processId);
                     var sw = Stopwatch.StartNew();
 
                     process = new Process()
                     {
-                        StartInfo = new ProcessStartInfo(fileName: executable)
+                        StartInfo = new ProcessStartInfo(executable ?? DefaultExecutable)
                         {
+                            WorkingDirectory = Program.ScriptDirectory,
                             UseShellExecute = false,
+                            CreateNoWindow = true,
                             RedirectStandardOutput = true,
                             RedirectStandardError = true,
-                            RedirectStandardInput = true,
+                            Arguments = args,
                         },
                     };
 
+                    process.StartInfo.Environment.Clear();
```

Comment:
Clearing the environment completely might cause problems. On Linux it tells you things like who the current user is, where to find executables, etc.

Reply:
i was thinking it would enhance security since the script would only have access to the data SLSKD set; sort of a 'least privileged' sort of thing.  in reality the shell process implicitly loads all environment variables on both Windows and Linux and this doesn't really do much unless someone is using an executable other than a shell.  i pulled it out for now.